GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -840,7 +840,8 @@ private void ShowLoading(bool sync = true)
         ///  Queries git for the new set of revisions and refreshes the grid.
         /// </summary>
         /// <exception cref="AggregateException"></exception>
-        public void PerformRefreshRevisions(Func<RefsFilter, IReadOnlyList<IGitRef>> getRefs = null)
+        // TODO: pass the cancellation token everywhere
```

Comment:
As `RefreshRevisions()` is called from many locations, why not use an internal token as currently in RevisionReader?

Reply:
I'm not sure what you mean. 

`RevisionReader` is instantiated and used internally by `RevisionGridControl`. The CT can come from outside (e.g., the `FormBrowse`).

Reply:
I suggest the CT is internal to `PerformRefreshRevisions()`, similar to how the `RevisionReader` currently handles it.
If one refresh is cancelled, a new is normally request immediately.